### PR TITLE
Refactor logic of deserialization of CommunicationIdentifierModel

### PR DESCRIPTION
**Pull request by ostoliarova-ee**
# Description

Refactored logic of deserialization of `CommunicationIdentifierModel` using `CommunicationIdentifierModelKind` and updated changelog files for respective modalities.

[User Story 3013988](https://skype.visualstudio.com/SPOOL/_workitems/edit/3013988): [SDK][Common][AllLangs] Prefer the "kind" property when deserializing the CommunicationIdentifier - follow up

### CODE REVIEW

File: sdk/communication/Azure.Communication.Chat/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## 1.2.0-beta.1 (Unreleased)
 
 ### Features Added
+- Added support for an optional `kind` property for `CommunicationIdentifierSerializer`.
```
Comment (petrsvihlik) [inline sdk/communication/Azure.Communication.Chat/CHANGELOG.md:6]: `CommunicationIdentifierSerializer` is an `internal` class that's not part of the public API and the customers have no knwledge about it. 
if you want to add a changelog, you may choose to add something like "Optimized the logic for deserializing types derived from the `CommunicationIdentifier`".

**Review by petrsvihlik — CHANGES_REQUESTED** [review]
I see that all the about-to-be-released packages are betas. figuring out the version works similarly to non-preview releases.
- check the latest released version, if there is none (first release), no need to bump anything
- consult your changes with semver
- set the new version of the package ignoring the "alpha/beta" tag (this is to ensure that once the package is out of beta, it'll still follow semver compared to the previously released stable version)

File: sdk/communication/Shared/src/CommunicationIdentifierSerializer.cs [hunk 2]
```diff
@@ -84,6 +59,26 @@ static void AssertMaximumOneNestedModel(CommunicationIdentifierModel identifier)
             }
         }
 
+        public static CommunicationIdentifierModelKind GetKind(CommunicationIdentifierModel identifier)
```
Comment (petrsvihlik) [inline sdk/communication/Shared/src/CommunicationIdentifierSerializer.cs:62]:
```suggestion
        private static CommunicationIdentifierModelKind GetKind(CommunicationIdentifierModel identifier)
```

**Review by petrsvihlik — CHANGES_REQUESTED** [review]

**Review by petrsvihlik — APPROVED** [review]

**Review by AikoBB — APPROVED** [review]

File: sdk/communication/Shared/src/CommunicationIdentifierSerializer.cs [hunk 3]
```diff
@@ -15,56 +15,31 @@ public static CommunicationIdentifier Deserialize(CommunicationIdentifierModel i
...
-                    AssertNotNull(teamsUser.IsAnonymous, nameof(teamsUser.IsAnonymous), nameof(MicrosoftTeamsUserIdentifierModel)),
-                    Deserialize(AssertNotNull(teamsUser.Cloud, nameof(teamsUser.Cloud), nameof(MicrosoftTeamsUserIdentifierModel))),
-                    rawId);
+                      AssertNotNull(user.UserId, nameof(user.UserId), nameof(MicrosoftTeamsUserIdentifierModel)),
+                      AssertNotNull(user.IsAnonymous, nameof(user.IsAnonymous), nameof(MicrosoftTeamsUserIdentifierModel)),
+                      Deserialize(AssertNotNull(user.Cloud, nameof(user.Cloud), nameof(MicrosoftTeamsUserIdentifierModel))),
+                      rawId);
             }
 
             return new UnknownIdentifier(rawId);
```
Comment (AikoBB) [inline sdk/communication/Shared/src/CommunicationIdentifierSerializer.cs:45]: for other platforms(java&js) we are checking for the non-nullability of rawId first, we have to unify this logic across all platforms

Comment (ostoliarova-msft) [reply sdk/communication/Shared/src/CommunicationIdentifierSerializer.cs:45]: @petrsvihlik @AikoBB here RawId is not used for creating `MicrosoftTeamsUserIdentifier` and `CommunicationUserIdentifier` checking it for null before trying to create any of these identifiers will break the existing logic

Comment (ostoliarova-msft) [reply sdk/communication/Shared/src/CommunicationIdentifierSerializer.cs:45]: @AikoBB  I've re-checked: there is already a validation of the rawId on the top of the method
https://github.com/Azure/azure-sdk-for-net/blob/8ce74c287bc545f1ea518a3364fe174d3d94e431/sdk/communication/Shared/src/CommunicationIdentifierSerializer.cs#L14
            `string rawId = AssertNotNull(identifier.RawId, nameof(identifier.RawId), nameof(CommunicationIdentifierModel));`

**Pull request merged**